pkg/operator/configobserver/node/observe_latency_profile.go

```diff
@@ -0,0 +1,110 @@
+package node
+
+import (
+	"reflect"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/klog/v2"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+// LatencyConfigProfileTuple is used to set arg value pairs for each worker latency profile
+type LatencyConfigProfileTuple struct {
+	// path to required argument in observed config
+	// eg. extendedArguments.node-monitor-grace-period from observed config
+	// is []string{"extendedArguments", "node-monitor-grace-period"}
+	ConfigPath   []string
+	ProfileValue map[configv1.WorkerLatencyProfileType]string
+}
+
+// NewLatencyProfileObserver is used to create ObserveConfigFunc that can be used with an configobservation controller to trigger
+// changes to different arg val pairs in observedConfig.* fields and update them on the basis of current worker latency profile
+func NewLatencyProfileObserver(latencyConfigs []LatencyConfigProfileTuple, conditionFn func() (bool, error)) configobserver.ObserveConfigFunc {
```

Comment:
can we name and document the conditionFn better?

Reply:
What name do you suggest @atiratree? I'd go away and update the doc first.

Reply:
something like `ShouldSuppressConfigUpdatesFn()`? But I would rather revisit this if we decide there should be a different implementation after discussing https://github.com/openshift/library-go/pull/1364#discussion_r868325737

Reply:
Updated it to: `ShouldSuppressConfigUpdatesFunc`.